Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -1226,7 +1259,13 @@ extension MacroApplication {
     else {
       return .notAMacro
     }
+
     do {
+      guard expandingFreestandingMacros.allSatisfy({ $0 != macro }) else {
```

Comment:
This implementation is quadratic in the depth of the stack.

Reply:
I used an array for the stack and a dictionary for memoizing macro cyclicalities in a previous commit, would that approach be better?

Reply:
I've force-pushed my previous approach, please have a look.

Reply:
We could make `expandingFreestandingMacros` a `Set` to avoid the quadratic complexity and use `!expandingFreestandingMacros.contains(macro)`

Reply:
> I do think this would read more intuitive as `!expandingFreestandingMacros.contains(macro)` though.

That means we'd be storing strings instead of Macro.Type in the stack?

Reply:
> That means we'd be storing strings instead of Macro.Type in the stack?

Oh, I forgot that metatypes don’t conform to `Comparable`. I’d still do `!expandingFreestandingMacros.contains({ $0 == macro })